web/impact/impact/v1/views/user_list_view.py

```diff
@@ -100,6 +106,44 @@ def _invalid_keys(self, user_keys):
         for key in set(user_keys) - set(ALL_USER_RELATED_KEYS):
             self.errors.append(INVALID_KEY_ERROR.format(key))
 
+    def _user_results(self, limit, offset):
+        queryset = User.objects.all()
+        updated_at_gt = self.request.query_params.get('updated_at__gt', None)
+        updated_at_lt = self.request.query_params.get('updated_at__lt', None)
+        if updated_at_gt or updated_at_lt:
+            queryset = _filter_profiles_by_date(
+                queryset,
+                updated_at_gt,
+                updated_at_lt)
+        return [
+            _serialize_user(user) for user in queryset[offset:offset + limit]]
+
+
+def _filter_profiles_by_date(queryset, updated_at_gt, updated_at_lt):
+    updated_at_gt = parse_date(updated_at_gt)
+    updated_at_lt = parse_date(updated_at_lt)
+    if updated_at_lt:
+        queryset = queryset.filter(
+            Q(expertprofile__updated_at__isnull=False) |
+            Q(entrepreneurprofile__updated_at__isnull=False) |
+            Q(memberprofile__updated_at__isnull=False)
+        ).exclude(
+            Q(expertprofile__updated_at__gte=updated_at_lt) |
+            Q(entrepreneurprofile__updated_at__gte=updated_at_lt) |
+            Q(memberprofile__updated_at__gte=updated_at_lt)
+        )
+    if updated_at_gt:
+        queryset.filter(
+            Q(expertprofile__updated_at__isnull=False) |
+            Q(entrepreneurprofile__updated_at__isnull=False) |
+            Q(memberprofile__updated_at__isnull=False)
+        ).exclude(
+            Q(expertprofile__updated_at__lte=updated_at_gt) |
+            Q(entrepreneurprofile__updated_at__lte=updated_at_gt) |
+            Q(memberprofile__updated_at__lte=updated_at_gt)
```

Comment:
I was seeing strange side-effects when inverting this clause - in particular it seemed to include items that were null etc